src/pinns_pde_solve.jl

```diff
@@ -99,6 +99,7 @@ struct PhysicsInformedNN{isinplace,C,T,P,PH,DER,PE,AL,ADA,LOG,K} <: AbstractPINN
   param_estim::PE
   additional_loss::AL
   adaptive_loss::ADA
+  adaptive_activation_function::ADF
```

Comment:
I don't think it needs to be added here. I think it just needs to be a property of the chain and added to the chain's weights list?

Reply:
Take a look at how the Flux functors are built: https://fluxml.ai/Flux.jl/stable/models/functors/

Reply:
> I think it just needs to be a property of the chain 

Could you elaborate a bit on how that would be done? 

I looked into how Flux functors are built and how they are used, would it be implemented like [this](https://github.com/FluxML/Flux.jl/blob/master/src/layers/basic.jl#L51)? It isn't exactly clear to me.

Reply:
One way you might implement this idea in Flux is by generating a `Flux.Chain` for the entire network that is a sequence of three building blocks, repeated:

```
Flux.Chain(
Dense(in, out, σ=identity; bias=true, init=nn_param_init),
AdaptiveActivation(n, a),
NonlinearActivation(nonlinearity),
)
```

The entire network would be those three layers repeated for how many hidden layers there are. Then you'd have to write a struct for `AdaptiveActivation` which would simply multiply the appropriate input by the `n a` current value in an elementwise fashion (very similar to the `Diagonal` implementation in `Flux`), and a `NonlinearActivation` struct which would have no trainable parameters but simply apply desired nonlinearity after the adaptive activation.  Basically in total you'd be recreating the `Dense` layer but with an extra elementwise operation in the middle.

This part of the Flux code would be good reference, since the building blocks you'd want to implement are similar to `Dense` and `Diagonal`, and you can see how the @functor macro gets used.
https://github.com/FluxML/Flux.jl/blob/ef04fda844ea05c45d8cc53448e3b25513a77617/src/layers/basic.jl#L82-L122

Interestingly the first paper's version is the hardest to implement here because you have to make sure that each of the AdaptiveActivations are utilizing the same value for `a` (this is known as weight tying or weight sharing).  That would go under the kind of stuff that is covered in this doc for Flux:
https://fluxml.ai/Flux.jl/stable/models/advanced/